source/adapters/level_zero/platform.cpp

```diff
@@ -304,14 +304,30 @@ ur_result_t ur_platform_handle_t_::populateDeviceCacheIfNeeded() {
 
   uint32_t ZeDeviceCount = 0;
   ZE2UR_CALL(zeDeviceGet, (ZeDriver, &ZeDeviceCount, nullptr));
+  uint32_t ZesDeviceCount = 0;
+  if (ZesDriver) {
+    ZE2UR_CALL(zesDeviceGet, (ZesDriver, &ZesDeviceCount, nullptr));
+  }
 
   try {
     std::vector<ze_device_handle_t> ZeDevices(ZeDeviceCount);
     ZE2UR_CALL(zeDeviceGet, (ZeDriver, &ZeDeviceCount, ZeDevices.data()));
+    std::vector<zes_device_handle_t> ZesDevices(ZesDeviceCount);
+    if (ZesDeviceCount > 0) {
+      ZE2UR_CALL(zesDeviceGet, (ZesDriver, &ZesDeviceCount, ZesDevices.data()));
+    }
 
     for (uint32_t I = 0; I < ZeDeviceCount; ++I) {
+      zes_device_handle_t zesDeviceHandle = nullptr;
+      // If the number of ZE Devices is greater than the number of ZES Devices,
+      // then a null sysman device is associated. The index of the ZE and ZES
+      // devices should match.
+      if (I < ZesDevices.size()) {
+        zesDeviceHandle = ZesDevices[I];
```

Comment:
do we need to match uuid here between core and sysman devices before association here? or you are not doing that for now since it was a very recent change in UMD?